### PR TITLE
FixedBufferAllocator.reset

### PR DESCRIPTION
**Pull request by Tetralux**

### CODE REVIEW

**Comment by Tetralux** [comment]
> 
> 
> Do we need a reset() function? `end_index` is considered to be a public field, and it's pretty simple concept to set it to zero. FixedBufferAllocator is a lightweight abstraction. I don't think "encapsulation" is valuable here.

It's a public field? Why is it not marked `pub` then?

If the field is in fact public, despite not being `pub`, then that needs making much more clear, and warrants explanation, _at least_ in the docs of `pub`.

I don't really get what you mean about encapsulation here. I'm just making clear that you are able to reset this kind of allocator, where before most people would have assumed that you cannot; especially since the field is non-`pub`; you don't want to rely on implementation details; if the name of field is changed then it breaks. It makes it more maintainable to have a simple `reset` procedure for it.

One use of this off the top of my head:
```zig
var buf: [1024]u8 = undefined;
var fba = std.heap.FixedBufferAllocator.init(buf[0..]);
var a = &fba.allocator;
while (true) : (fba.reset()) {
    // do some work, use the result, and throw it all away at the end by resetting.
}
```

File: std/heap.zig [hunk 1]
```diff
@@ -480,6 +480,10 @@ pub const FixedBufferAllocator = struct {
     fn shrink(allocator: *Allocator, old_mem: []u8, old_align: u29, new_size: usize, new_align: u29) []u8 {
         return old_mem[0..new_size];
     }
+
+    pub inline fn reset(self: *FixedBufferAllocator) void {
```
[the remark this answers is not in the document]
Comment (Tetralux) [reply std/heap.zig]: Done.

**Comment by andrewrk** [comment]
> It's a public field? Why is it not marked `pub` then?

Fair question. `pub` for fields is having a bit of an existential crisis at the moment. At this point I'm more likely to remove that syntax than make it do anything.

**Comment by Tetralux** [comment]
Cheers!

**Pull request merged**